### DISCUSSION TITLE
Why are `getCoreRowModel`, `onStateChange` and `renderFallbackValue` required?

### DISCUSSION THREAD

**Discussion by AlexChadwickP** [KevinVandy/material-react-table#303] [Q&A] [answered]
I'm sure there is a good reason for these props to be required. But if I'm able to create a material react table component just fine without specifying these attributes (except for some linting errors), it poses the question: are these _actually_ required? Or are they just another good practice or convention that should be followed?

If it's the latter, then in my opinion (and I'm open to other perspectives) they shouldn't be a required prop, but we should stress somewhere in the documentation the importance of configuring these correctly.

Mainly because if, for example, we're prototyping something and we just want to knock up a simple table, we might not be concerned about the best way of doing things, but just about doing things quickly.

What are peoples thoughts on this?

**Comment by KevinVandy** [comment]
They are not required in MRT, they just required internally in the lib for react table. Where are you seeing this?

**Reply by KevinVandy** [reply]
If you are looking at the [props table](https://www.material-react-table.com/docs/api/props), you should see that only the `columns` and `data` props are required. And this is also reflected in the TypeScript types. I don't know how you would be seeing this. `onStateChange` actually does not exist as a prop on MaterialReactTable.

**Reply by AlexChadwickP** [reply]
I'm now thinking this is a bug with my IDE as I opened my fork of MRT and the I opened my work project, after restarting it the error disappeared. I thought it was odd that the error wasn't stopping me from compiling the application!

**Comment by dnjsgur0629** [comment]
I'm having the same problem as written here. But restarting the IDE didn't solve it. I am using `Webstorm 2022.1`.

Create a `BaseTable` component as follows:
```ts
import React from 'react'
import MaterialReactTable, { MaterialReactTableProps } from 'material-react-table'
import { MyColumnDef } from '@myTypes
...

interface Props<TData> extends Omit<MaterialReactTableProps<TData>, 'columns'> {
  columns: MyColumnDef<TData>[]  // I customized MRT_ColumnDef, but it doesn't seem to be an important part in this issue.
  columnPinning?: {
    left?: Array<string>
    right?: Array<string>
  }
  columnsToSum?: Array<keyof TData>
}

export const BaseTable = <TData,>({
  columns,
  data,
  columnPinning,
  columnsToSum,
  initialState,
  ...otherProps
}: Props<TData>) => {
  ...
}
```
And if I use `BaseTable` somewhere in my project, the following Warning occurs.
![image](https://user-images.githubusercontent.com/65391273/217419756-7c2c1f0e-c1e1-46ac-aeea-36a35570e24f.png)

Unfortunately I failed to reproduce this issue in codesandbox. No warning appeared in codesandbox. Is there something wrong with my Webstorm setting? If you have any ideas, please let me know. thank you

**Reply by KevinVandy** [reply]
I'm guessing it's a web storm specific thing. My guess to get around it would be to just wrap a Partial around MaterialReactTableProps

**Reply by dnjsgur0629** [reply]
Wrapping a partial around MaterialReactTableProps seems like a good idea. Now I am free from that warning. thank you.

**Comment by KevinVandy** [comment, marked as answer]
If you are using Webstorm and this happens: Just wrap a Partial around MaterialReactTableProps like this:
`Partial<MaterialReactTableProps>`
